src/Workspaces/Core/Portable/Storage/SQLite/Interop/SqlConnection.cs

```diff
@@ -67,24 +77,34 @@ public static SqlConnection Create(IPersistentStorageFaultInjector faultInjector
 
             Contract.ThrowIfNull(handle);
 
-            raw.sqlite3_busy_timeout(handle, (int)TimeSpan.FromMinutes(1).TotalMilliseconds);
-
-            var queryToStatement = new Dictionary<string, SqlStatement>();
-            return new SqlConnection(handle, faultInjector, queryToStatement);
+            try
+            {
+                raw.sqlite3_busy_timeout(handle, (int)TimeSpan.FromMinutes(1).TotalMilliseconds);
+                return new SqlConnection(handle, faultInjector, queryToStatement);
+            }
+            catch
+            {
+                // If we failed to create connection, ensure that we still release the sqlite
+                // handle.
+                raw.sqlite3_close(handle);
```

Comment:
❔ Why not `handle.Dispose()`?

Reply:
I don't know what the semantics of that are.  I know the semantics of sqlite3_close.  :)   i prefer to use the sqlite3_xxx apis because they have very good actual docs over in sqlite-land that i can get very crisp semantics on.